### ISSUE TITLE
get datatypes into Galaxy

### ISSUE THREAD

**Issue by bgruening** [workflow4metabolomics/xcms#56]
@lecorguille if we hurry up we might get the datatypes into 17.05 and ready for GCC :)
The unzip datatype can be removed isn't it? This should now be supported by Galaxy naively.

**Comment by lecorguille** [comment]
I'm not fond of gather all the Datatype into the core. I didn't want to pollute the `datatypes_conf.xml.sample`

About the unzip datatype, I believe that we need to train a little people to the new system before removing the zip. And no, Galaxy don't deal with the zip natively as with the unzip datatype. This one come with a sniffer, John didn't want to hear about :/

What is the hurry for you to push them to the core?
Is it for new contributors? Because, shiped datatype are a little old fashion?

**Comment by bgruening** [comment]
datatypes in the TS has a lot of disadvantages

 * they are not versioned
 * you pollute every repository with these files and Galaxy will load all of the
 * reducing startup time
 * they are not tested, so they potentially destroy the sniffing order (order is crucial)

No pressure, just saw the possibility to get it into 17.05 for GCC :)

**Comment by pcm32** [comment]
We would really like this to happen! I'm currently trying W4M's XCMS on our deployment, and we need to manually add datatypes to the docker images. It would be great if they would be present already on Galaxy. Same with the datatypes on CAMERA. Thanks for all the great work @lecorguille and colleagues.

**Comment by lecorguille** [comment]
How to you exactly install the tools within the docker image?
Because, if you use the [Ansible roles](https://github.com/workflow4metabolomics/w4m-docker/blob/master/Dockerfile#L20) as it's used by the Galaxy Flavor, it works!

**Comment by pcm32** [comment]
Yes, our tool installation process is a bit different from normal in that we need to set up destinations for tools and we don't install any software dependencies (I learn now that shed-install can deal with something like this), so we simply copy tools directories to the correct place inside Galaxy. I presume that through shed-install I could get the tools and related datatypes (as repo dependencies), avoid installing dependencies, but I still I would need to write the job_conf files. I don't think you can tell shed-install to add destinations for tools quite yet (might be wrong). Good to know in any case, thanks! I have sorted it for now anyway, but still think it would be nice to have datatypes upstream.

**Comment by yguitton** [comment]
may be liked to 
https://github.com/workflow4metabolomics/w4m-vm/issues/30

**Comment by bernt-matthias** [comment]
Just had the problem that several versions of the datatypes were installed. These caused a conflict which had the result that the datatypes were unavailable. Solved it by uninstalling the older one. 

But I guess this just underlines that it might be a good idea to bring them to the Galaxy core.

**Comment by bernt-matthias** [comment]
More precisely the old tool versions seem to become dysfunctional when new datatypes are installed (one datatype was replaced by another). Just looking into this.

**Comment by lecorguille** [comment]
Hum ... It's been a while since I update my datasets ...

**Comment by bernt-matthias** [comment]
For the Rdata datatypes one might even skip the python part. Instead of: 

`<datatype extension="rdata.msnbase.raw" type="galaxy.datatypes.rdata_xcms_datatype:RdataMsnbaseRaw" mimetype="application/x-gzip" subclass="true" display_in_upload="true" />`

one could simply:

`<datatype extension="rdata.msnbase.raw" type="galaxy.datatypes.binary:RData" mimetype="application/x-gzip" subclass="true" display_in_upload="true" />`

**Comment by RJMW** [comment]
I have started to include the datatypes for xcms and camera in galaxy core.

See here: https://github.com/galaxyproject/galaxy/pull/9470

@lecorguille @bernt-matthias @bgruening  can you briefly review the PR?

**Comment by hechth** [comment]
@bgruening @lecorguille  I think this can be closed? AFAIK all datatypes have been migrated into Galaxy core?
